doc/source/notebooks/intro_to_gpflow2.pct.py

```diff
@@ -223,52 +223,68 @@ def simple_training_loop(model: gpflow.models.SVGP, epochs: int = 1, logging_epo
+from gpflow.monitor import (
+    ImageToTensorBoard,
+    ModelToTensorBoard,
+    ScalarToTensorBoard,
+    MonitorCollection,
+)
+
+samples_input = np.linspace(0, 10, 100).reshape(-1, 1)
+
+
+def plot_model(fig, ax):
+    mean, var = model.predict_f(samples_input)
+    num_samples = 10
+    samples = model.predict_f_samples(samples_input, num_samples)
+    ax.plot(samples_input, mean, "C0", lw=2)
+    ax.fill_between(
+        samples_input[:, 0],
+        mean[:, 0] - 1.96 * np.sqrt(var[:, 0]),
+        mean[:, 0] + 1.96 * np.sqrt(var[:, 0]),
+        color="C0",
+        alpha=0.2,
+    )
+    ax.plot(X, Y, "kx")
+    ax.plot(samples_input, samples[:, :, 0].numpy().T, "C0", linewidth=0.5)
+    ax.set_ylim(-2.0, +2.0)
+    ax.set_xlim(0, 10)
+
 
-samples_input = to_default_float(np.linspace(0, 10, 100).reshape(100, 1))
+output_logdir = enumerated_logdir()
+model_task = ModelToTensorBoard(output_logdir, model)
+image_task = ImageToTensorBoard(output_logdir, plot_model, "samples_image")
+elbo_task = ScalarToTensorBoard(output_logdir, lambda data=None: model.elbo(data).numpy(), "elbo")
+monitor = MonitorCollection([model_task, image_task, elbo_task])
 
 
 def monitored_training_loop(
-    model: gpflow.models.SVGP,
-    logdir: str,
-    epochs: int = 1,
-    logging_epoch_freq: int = 10,
-    num_samples: int = 10,
+    epochs: int = 1, logging_epoch_freq: int = 10,
 ):
-    summary_writer = tf.summary.create_file_writer(logdir)
     tf_optimization_step = tf.function(optimization_step)
     batches = iter(train_dataset)
 
-    with summary_writer.as_default():
-        for epoch in range(epochs):
-            for _ in range(num_batches_per_epoch):
-                tf_optimization_step(model, next(batches))
-
-            epoch_id = epoch + 1
-            if epoch_id % logging_epoch_freq == 0:
-                tf.print(f"Epoch {epoch_id}: ELBO (train) {model.elbo(data)}")
-
-                mean, var = model.predict_f(samples_input)
-                samples = model.predict_f_samples(samples_input, num_samples)
-                fig = plotting_regression(X, Y, samples_input, mean, var, samples)
+    for epoch in range(epochs):
+        for _ in range(num_batches_per_epoch):
+            batch = next(batches)
+            tf_optimization_step(model, batch)
 
-                summary_matplotlib_image(dict(model_samples=fig), step=epoch)
-                tf.summary.scalar("elbo", data=model.elbo(data), step=epoch)
-                tf.summary.scalar("likelihood/variance", data=model.likelihood.variance, step=epoch)
-                tf.summary.scalar("kernel/lengthscales", data=model.kernel.lengthscales, step=epoch)
-                tf.summary.scalar("kernel/variance", data=model.kernel.variance, step=epoch)
+        epoch_id = epoch + 1
+        if epoch_id % logging_epoch_freq == 0:
+            tf.print(f"Epoch {epoch_id}: ELBO (train) {model.elbo(data)}")
+            monitor(epoch, data=batch)
```

Comment:
why not call it every step (monitor should be handling the only-every-nth-step/epoch)?

Reply:
also, how about showing how you can compile it ? (or at least mention it here and refer to the main monitoring notebook)?